src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -0,0 +1,191 @@
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtWidgets import *
+from software.py_constants import *
+from proto.import_all_protos import *
+import software.thunderscope.common.common_widgets as common_widgets
+from software.thunderscope.constants import Colors, ERROR_CODE_MESSAGES
+
+
+class RobotInfo(QWidget):
+
+    toggle_one_connection_signal = QtCore.pyqtSignal(int, int)
+
+    def __init__(self, robot_id, load_fullsystem):
+        """
+        Initialize a single robot's info widget
+        """
+
+        super().__init__()
+
+        self.robot_id = robot_id
+
+        # when this robot has a battery warning, this is set to True
+        # which prevents spamming the same battery warning
+        # set back to False if battery is back above warning level
+        self.battery_warning_disabled = False
+
+        self.layout = QHBoxLayout()
+
+        self.status_layout = QVBoxLayout()
+
+        # Battery Bar
+        self.battery_layout = QHBoxLayout()
+        self.battery_progress_bar = common_widgets.ColorProgressBar(
+            MIN_BATTERY_VOLTAGE - 3, MAX_BATTERY_VOLTAGE
```

Comment:
nit: Can you either move the -3 to a variable right above this, or comment on why you decided to add an offset?